docs/developer-guide/etendo-classic/how-to-guides/how-to-create-junit-testcases.md

```diff
@@ -0,0 +1,525 @@
+
+```java
+ 
+@Test
+public void testMatchesObject() {
+  JSONObject json1 = new JSONObject(Map.of("p1", 1, "p2", "abcd"));
+  JSONObject json2 = new JSONObject(Map.of("p1", 1, "p2", "abcd"));
+  JSONObject json3 = new JSONObject(Map.of("p2", "abcd", "p1", 1));
+  JSONObject json4 = new JSONObject(Map.of("p1", 1, "p2", "efgh"));
+  JSONObject json5 = new JSONObject(Map.of("p1", 1));
+  JSONObject json6 = new JSONObject(Map.of("p1", 1, "p2", "abcd", "p3", "abcd"));
+  JSONObject json7 = new JSONObject(Map.of("p1", greaterThan(0), "p2", "abcd"));
+ 
+  assertThat("JSON object match", json1, matchesObject(json2));
+  assertThat("JSON object match", json1, matchesObject(json3));
+  assertThat("JSON object does not match", json1, not(matchesObject(json4)));
+  assertThat("JSON object match", json1, matchesObject(json5));
+  assertThat("JSON object does not match", json1, not(matchesObject(json6)));
+  assertThat("JSON object match", json1, matchesObject(json7));
+}
+```
+
+####  hasItems
+
+Used to match the items of a JSONArray. This matcher can be used with two
+different kind of arguments.
+
+If an array of objects is passed, then it matches when the examined JSONArray
+contains all the received objects. The order of the objects is not taken into
+account.
+
+```java   
+ 
+@Test
+public void testHasItems() {
+  JSONObject json1 = new JSONObject(Map.of("p1", 1, "p2", "abcd"));
+  JSONObject json2 = new JSONObject(Map.of("p2", 2, "p3", "efgh"));
+  JSONArray array = new JSONArray();
+  array.put(json1);
+  array.put(json2);
+  array.put(5);
+  JSONObject json3 = new JSONObject(Map.of("p2", 2, "p3", startsWith("e")));
+  JSONObject json4 = new JSONObject(Map.of("p2", 2, "p3", "ijkl"));
+ 
+  assertThat("JSON array has items", array, hasItems(5, json3));
+  assertThat("JSON array does not have items", array, not(hasItems(4, json3)));
+  assertThat("JSON array does not have items", array, not(hasItems(json4)));
+}
+```
+
+It also supports receiving an array of  Hamcrest  matchers. In that case, then
+it matches when the examined JSONArray matches with all the received matchers.
+
+```java
+ 
+@Test
+public void testHasItems() {
+  JSONObject json1 = new JSONObject(Map.of("p1", 1, "p2", "abcd"));
+  JSONObject json2 = new JSONObject(Map.of("p2", 2, "p3", "efgh"));
+  JSONArray array = new JSONArray();
+  array.put(json1);
+  array.put(json2);
+  array.put(5);
+  JSONObject json3 = new JSONObject(Map.of("p2", 2, "p3", startsWith("e")));
+  JSONObject json4 = new JSONObject(Map.of("p1", 1));
+  JSONObject json5 = new JSONObject(Map.of("p2", 2, "p3", "ijkl"));
+ 
+  assertThat("JSON array has items", array, hasItems(greaterThan(4), equal(json3)));
+  assertThat("JSON array has items", array, hasItems(greaterThan(4), matchesObject(json4)));
+  assertThat("JSON array does not have items", array, not(hasItems(greaterThan(5))));
+  assertThat("JSON array does not have items", array, not(hasItems(equal(json5))));
+}
+```
+
+###  Ant Test Tasks
```

Comment:
These Ant tasks are executed from Gradle. 
I would change the title to "Test Task" only.